src/PMGGenverterStatus.cpp

```diff
@@ -0,0 +1,112 @@
+#include <power_whisperpower/PMGGenverterStatus.hpp>
+
+using namespace power_whisperpower;
+
+using std::endl;
+
+std::string to_string(PMGGenverterStatus::InverterAlarm inverter_alarm)
+{
+    switch (inverter_alarm) {
+        case PMGGenverterStatus::AC_SHORT_CIRCUIT:
+            return "AC_SHORT_CIRCUIT";
+        case PMGGenverterStatus::AC_OVER_LOAD:
+            return "AC_OVER_LOAD";
+        case PMGGenverterStatus::AC_UNDER_VOLTAGE:
+            return "AC_UNDER_VOLTAGE";
+        case PMGGenverterStatus::OVER_TEMPERATURE:
+            return "OVER_TEMPERATURE";
+        case PMGGenverterStatus::LOST_REMOTE_COMMUNICATION:
+            return "LOST_REMOTE_COMMUNICATION";
+        case PMGGenverterStatus::PARAMETERS_ERROR:
+            return "PARAMETERS_ERROR";
+        case PMGGenverterStatus::INVERTER_VDRIVERLOW:
+            return "INVERTER_VDRIVERLOW";
+        case PMGGenverterStatus::ALTERNATOR_MISSING_PHASE:
+            return "ALTERNATOR_MISSING_PHASE";
+        case PMGGenverterStatus::RPM_OVER_SPEED:
+            return "RPM_OVER_SPEED";
+    }
+    throw std::runtime_error("invalid inverter alarm received");
+}
+
+std::string to_string(PMGGenverterStatus::EngineAlarm engine_alarm)
+{
+    switch (engine_alarm) {
```

Comment:
These are *bitfields*. You can have more than one of the values set at the same time.